server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -488,22 +491,34 @@ private void setSlowWriteLoggingThreshold(TimeValue slowWriteLoggingThreshold) {
         this.slowWriteLoggingThreshold = slowWriteLoggingThreshold;
     }
 
-    private static String getManifestFileName(long term, long version) {
-        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/manifest/manifest_2147483642_2147483637_456536447
-        return String.join(DELIMITER, getManifestFileNamePrefix(term, version), RemoteStoreUtils.invertLong(System.currentTimeMillis()));
+    static String getManifestFileName(long term, long version, boolean committed) {
+        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/manifest/manifest_2147483642_2147483637_1_21473637_456536447
+        return String.join(
+            DELIMITER,
+            getManifestFileNamePrefix(term, version, committed),
+            RemoteStoreUtils.invertLong(MANIFEST_CODEC_VERSION),
+            RemoteStoreUtils.invertLong(System.currentTimeMillis())
+        );
     }
 
-    private static String getManifestFileNamePrefix(long term, long version) {
-        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/manifest/manifest_2147483642_2147483637
-        return String.join(DELIMITER, MANIFEST_PATH_TOKEN, RemoteStoreUtils.invertLong(term), RemoteStoreUtils.invertLong(version));
+    private static String getManifestFileNamePrefix(long term, long version, boolean committed) {
```

Comment:
can we combine this with above method or this is used somewhere else as well?

Reply:
It is not being used at other placed. Combined it.